src/Sarif/Visitors/RebaseUriVisitor.cs

```diff
@@ -34,6 +34,11 @@ public RebaseUriVisitor(string uriBaseId, Uri baseUri, bool rebaseRelativeUris =
 
         public override ArtifactLocation VisitArtifactLocation(ArtifactLocation node)
         {
+            if (node.Uri == null)
+            {
+                return node;
+            }
+
```

Comment:
the "fix". Everything else you see in this pr is tests and related infra

Reply:
more explanation: ArtifactLocation could be just have a single property "index" wich indexes into run.artifacts.

---
In reply to: [298444180](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298444180) [](ancestors = 298444180)

Reply:
That's so bizarre. I wasted time because CodeFlow colored everything from here down to the end of the file. It's just these three lines that are new. Change looks good.

---
In reply to: [298444310](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298444310) [](ancestors = 298444310,298444180)

Reply:
oh! strange! thanks though for bonus inputs :) i will make the changes

---
In reply to: [298658405](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298658405) [](ancestors = 298658405,298444310,298444180)

Reply:
PS: i see the strange codeflow behaviour as well (will show it to Birendra)

---
In reply to: [298689012](https://github.com/microsoft/sarif-sdk/pull/1549#discussion_r298689012) [](ancestors = 298689012,298658405,298444310,298444180)